applications/GeoMechanicsApplication/tests/cpp_tests/test_newmark_quasistatic_U_Pw_scheme.cpp

```diff
@@ -0,0 +1,126 @@
+// KRATOS___
+//     //   ) )
+//    //         ___      ___
+//   //  ____  //___) ) //   ) )
+//  //    / / //       //   / /
+// ((____/ / ((____   ((___/ /  MECHANICS
+//
+//  License:         geo_mechanics_application/license.txt
+//
+//  Main authors:    Richard Faasse
+//
+
+#include "testing/testing.h"
+
+#include "custom_strategies/schemes/newmark_quasistatic_U_Pw_scheme.hpp"
+#include "spaces/ublas_space.h"
+#include "test_utilities/spy_element.h"
+#include "test_utilities/spy_condition.h"
+
+namespace Kratos::Testing {
+
+using namespace Kratos;
+using SparseSpaceType = UblasSpace<double, CompressedMatrix, Vector>;
+using LocalSpaceType = UblasSpace<double, Matrix, Vector>;
+
+class NewmarkQuasistaticUPwSchemeTester {
+public:
+    Model mModel;
+    NewmarkQuasistaticUPwScheme<SparseSpaceType, LocalSpaceType> mScheme =
+        CreateValidScheme();
+
+    NewmarkQuasistaticUPwSchemeTester()
+    {
+        CreateValidModelPart();
+    }
+
+    NewmarkQuasistaticUPwScheme<SparseSpaceType, LocalSpaceType> CreateValidScheme() const
+    {
+        NewmarkQuasistaticUPwScheme<SparseSpaceType, LocalSpaceType> result(0.25, 0.5, 0.75);
+        return result;
+    }
+
+    void CreateValidModelPart()
+    {
+        auto& result = mModel.CreateModelPart("dummy", 2);
+        result.AddNodalSolutionStepVariable(TEMPERATURE);
+        result.AddNodalSolutionStepVariable(VELOCITY);
+        result.AddNodalSolutionStepVariable(ACCELERATION);
+        result.AddNodalSolutionStepVariable(DT_WATER_PRESSURE);
+        result.AddNodalSolutionStepVariable(DISPLACEMENT);
+        result.AddNodalSolutionStepVariable(WATER_PRESSURE);
+
+        auto p_node = result.CreateNewNode(0, 0.0, 0.0, 0.0);
+        p_node->AddDof(DISPLACEMENT_X);
+        p_node->AddDof(DISPLACEMENT_Y);
+        p_node->AddDof(DISPLACEMENT_Z);
+        p_node->AddDof(WATER_PRESSURE);
+        result.GetProcessInfo()[DELTA_TIME] = 4.0;
+
+        p_node->FastGetSolutionStepValue(VELOCITY, 1) =
+            Kratos::array_1d<double, 3>{1.0, 2.0, 3.0};
+        p_node->FastGetSolutionStepValue(ACCELERATION, 1) =
+            Kratos::array_1d<double, 3>{4.0, 5.0, 6.0};
+        p_node->FastGetSolutionStepValue(DISPLACEMENT, 1) =
+            Kratos::array_1d<double, 3>{7.0, 8.0, 9.0};
+
+        p_node->FastGetSolutionStepValue(WATER_PRESSURE, 1) = 1.0;
+        p_node->FastGetSolutionStepValue(WATER_PRESSURE) = 2.0;
+    }
+
+    ModelPart& GetModelPart()
+    {
+        return mModel.GetModelPart("dummy");
+    }
+};
+
+KRATOS_TEST_CASE_IN_SUITE(CheckScheme_ReturnsZeroForValidScheme, KratosGeoMechanicsFastSuite)
+{
+    NewmarkQuasistaticUPwSchemeTester tester;
+    KRATOS_EXPECT_EQ(tester.mScheme.Check(tester.GetModelPart()), 0);
+}
+
+KRATOS_TEST_CASE_IN_SUITE(InitializeUPWScheme_SetsTimeFactors, KratosGeoMechanicsFastSuite)
+{
+    NewmarkQuasistaticUPwSchemeTester tester;
+    tester.GetModelPart().GetProcessInfo()[DELTA_TIME] = 4.0;
```

Comment:
I think this statement is redundant. (This is already done by `CreateValidModelPart`, which is called when constructing a `NewmarkQuasistaticUPwSchemeTester` object.)